src/fparser/two/parser.py

```diff
@@ -0,0 +1,262 @@
+    for cls_member in all_cls_members:
+        if cls_member[1].__module__ == module_name:
+            module_cls_members.append(cls_member)
+    return module_cls_members
+
+
+class ParserFactory(object):
+    '''Creates a parser suitable for the specified fortran standard'''
+
+    def create(self, std=None):
+        '''Creates a class hierarchy suitable for the specified fortran
+        standard
+
+        :param std string: the fortran standard. Choices are 'f2003' or
+                           'f2008'. 'f2003' is the default.
+        :return: a Program class (not object) for use with the fortran reader
+        :rtype: :py:class:`fparser.two.Fortran2003.Program`
+
+        For example:
+
+        >>> from fparser.two.parser import ParserFactory
+        >>> f2003_parser = ParserFactory().create()
+        >>> f2003_parser = ParserFactory().create(std='f2003')
+        >>> f2008_parser = ParserFactory().create(std='f2008')
+
+        '''
+        # find all relevant classes in our Fortran2003 file as we
+        # always need these
+        from fparser.two import Fortran2003
+        f2003_cls_members = get_module_classes(Fortran2003)
+        if not std:
+            # default to f2003
+            std = "f2003"
+        if std == "f2003":
+            # we already have our required list of classes so call _setup
+            # to setup our class hierarchy
+            self._setup(f2003_cls_members)
+            # the class hierarchy has been set up so return the top
+            # level class that we start from when parsing fortran code
+            return Fortran2003.Program
+        elif std == "f2008":
+            # we need to find all relevent classes in our Fortran2003
+            # and Fortran2008 files and then ensure that where classes
+            # have the same name we return the Fortran2008 class
+            # i.e. where Fortran2008 extends Fortran2003 we return
+            # Fortran2008.
+            # First find all Fortran2008 classes
+            from fparser.two import Fortran2008
+            f2008_cls_members = get_module_classes(Fortran2008)
+            # next add in Fortran2003 classes if they do not already
+            # exist as a Fortran2008 class
+            f2008_class_names = [i[0] for i in f2008_cls_members]
+            for local_cls in f2003_cls_members:
+                if local_cls[0] not in f2008_class_names:
+                    f2008_cls_members.append(local_cls)
+            # we now have our required list of classes so call _setup
+            # to setup our class hierarchy
+            self._setup(f2008_cls_members)
+            # the class hierarchy has been set up so return the top
+            # level class that we start from when parsing fortran
+            # code. Fortran2008 does not extend the top level class so
+            # we return the Fortran2003 one.
+            return Fortran2003.Program
+        else:
+            print ("unsupported standard {0}".format(std))
```

Comment:
Exceptions need to be sorted in fparser2 in general and we have an issue to look at this #52. Therefore I decided to simply exit for the time being, rather than spending time creating a new exception that we would need to change. Is that a good enough argument or would you like me to add an exception in any case?

Reply:
That's a good enough argument but please add a TODO linking to #52.